### PR TITLE
i3config syntax with better keyword/type distinction

### PR DESCRIPTION
**Pull request by litoj**
This syntax uses a couple more colour highlights to match the actual function of used keywords in the config file. This way it is easier to figure out what does what.
- distinctly highlights key combos to clearly differentiate it from the rest of the command
- all keywords that can be used as command line commands are highlighted as Functions
- conditions are now properly highlighted with all operators clearly visible
+ some bug fixes

file updated at: https://github.com/JosefLitos/vim-i3config

### CODE REVIEW

**Comment by hiqua** [comment]
Thanks! Will review.

File: runtime/syntax/i3config.vim [hunk 1]
```diff
@@ -1,9 +1,9 @@
 " Vim syntax file
 " Language: i3 config file
 " Original Author: Mohamed Boughaba <mohamed dot bgb at gmail dot com>
-" Maintainer: Quentin Hibon (github user hiqua)
-" Version: 0.4
-" Last Change: 2022 Jun 05
+" Maintainer: Josef Litoš (github JosefLitos)
```
Comment (hiqua) [inline runtime/syntax/i3config.vim]: Sure about that? :smile:

File: runtime/syntax/i3config.vim [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+" Version: 4.20
```
Comment (hiqua) [inline runtime/syntax/i3config.vim]: I guess we can remove the version line altogether, it makes little sense IMHO.

File: runtime/syntax/i3config.vim [hunk 3]
```diff
@@ -17,39 +17,39 @@ endif
 
 scriptencoding utf-8
 
+" Error
+syn match i3ConfigError /.*/
```
Comment (hiqua) [inline runtime/syntax/i3config.vim]: I've removed that already in a separate commit, can you remove it as well? It poses problem whenever a new keyword is introduced e.g. in sway, and we're not fully up to date, which is bound to happen regularly.

**Review by hiqua — CHANGES_REQUESTED** [review]
I know it's a long shot, but, before I deep-dive into the code to review it, any chance you could try to reduce the diff? That would make my life way easier.

Otherwise I'll make do with this, but it might take me more time.

**Comment by JosefLitos** [comment]
I will get to it as soon as I can (in August).

**Comment by JosefLitos** [comment]
I reorganized it a little. But the color binds I left untouched as I find it a little clearer this way.

**Comment by hiqua** [comment]
Thanks! I'll have a look.

**Comment by hiqua** [comment]
Hi @JosefLitos, sorry I took some time, and now your PR seems to have conflicts and failing the CI, can you rebase it before I delve into it? I promise I won't be that long after that.

**Comment by JosefLitos** [comment]
@hiqua I am a little afraid I did the rebase procedure incorrectly as all the new commits have been made. But it shouldn't interfere with master anymore.

**Pull request closed without merging**